lib/emojibot/bot.ex

```diff
@@ -2,9 +2,20 @@ defmodule Emojibot.Bot do
   # Inclusions
   use Slack
 
+  # Constants
+  @message_options %{
+    icon_url: "https://avatars.slack-edge.com/2017-09-16/242182638770_50d22a57a3544dd0bbb7_192.png",
+    username: "emojibot"
+  }
+
   def handle_event(%{name: name, type: "emoji_changed", subtype: "add"}, slack, state) do
-    emoji_channel = Application.get_env(:emojibot, :emoji_channel)
-    send_message(":#{name}: (`#{name}`)", emoji_channel, slack)
+    emoji_channel_id =
+      :emojibot
+      |> Application.get_env(:emoji_channel)
+      |> Slack.Lookups.lookup_channel_id(slack)
+
+    %{"ok" => true, "message" => %{"ts" => ts}} = Slack.Web.Chat.post_message(emoji_channel_id, ":#{name}:", @message_options)
+    %{"ok" => true} = Slack.Web.Chat.post_message(emoji_channel_id, "`#{name}`", Map.put(@message_options, :thread_ts, ts))
```

Comment:
what's the purpose to pattern match on last response ?

What happened if something doesn't pattern match in these two lines ?

Reply:
> what's the purpose to pattern match on last response ?

To make sure the message was correctly sent.

> What happened if something doesn't pattern match in these two lines ?

It will raise an error.

I’ll be honest, the error handling is poorly (ie. not) implemented. But I want to let it raise errors and not just silently swallow them 😄